_applications/wine-map.adoc

```diff
@@ -0,0 +1,121 @@
+The spark cluster takes the wine review data and calculates the average number of points each country has scored (this is calculated using different wines from the same country).
+
+pass:[<img src="/assets/wine-map/wine-map-architecture.png" alt="Wine Map Architecture" class="img-responsive" width="400" height="600">]
+
+
+[[installation]]
+== Installation
+
+First, you must download the oc client tools this will enable you to start up your own OpenShift cluster. These can be downloaded at: https://www.OpenShift.org/download.html. There is also documentation about how to use the oc tools here: https://docs.OpenShift.org/latest/cli_reference/get_started_cli.html.
+Now, lets start putting the application together you need to start up your own OpenShift cluster, the following command will allow you to do this:
+
+....
+oc cluster up
+....
+
+The next step is to create a project for the wine map application to reside in, here is a link to some documentation on OpenShift projects: https://docs.OpenShift.com/container-platform/3.7/dev_guide/projects.html
+Type in the following command to create your project:
+
+....
+oc new-project winemap-namespace
+....
+
+After you have created an OpenShift cluster and a project to deploy containers into the next step is to deploy a PostgreSQL database.
+This can be done easily using the PostgreSQL template provided by OpenShift with parameters stating the username, password and database name.
+For this example just copy the command below:
+
+....
+oc new-app --template=PostgreSQL-persistent -p PostgreSQL_USER=username -p PostgreSQL_PASSWORD=password -p PostgreSQL_DATABASE=wineDb
+....
+
+The next step is to load the data into the database, for this you will need to clone this repository https://github.com/radanalyticsio/winemap-data-loader.
+We will then create the template for the Kubernetes job (here is a link to explain how templates work in more detail: https://docs.OpenShift.org/latest/dev_guide/templates.html).
+The template references an image which executes a python script to load the data. The command you need to use is:
+....
+oc create -f wine-data-loader.yaml
+....
+
+We then need to set up a secret to pass the database the username, password and database name.
+Secrets are used in OpenShift to allow a more secure means of passing important data,
+as the data is encrypted in the secret file (https://docs.OpenShift.com/container-platform/3.5/dev_guide/secrets.html). Here is the command used to create the secret file in the project:
+
+....
+oc create -f secret.yaml
+....
+
+We then create the Kubenettes job to load the data with the following command:
+
+....
+oc new-app --template=wine-data-loader
+....
+
+The next half of the instructions describes how to create the flask server and spark cluster for processing and displaying the map.
+First of all you must create the oshinko resources in your OpenShift cluster with the following command:
```

Comment:
we should just link back to the get started info from above, the concern here is that we don't want to duplicate the install instructions in case they ever change.

Reply:
changed